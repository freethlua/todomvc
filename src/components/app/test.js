const { h } = require('preact');
const { shallow, deep } = require('preact-render-spy');
const chain = require('chain-free');
const td = require('testdouble');
const Component = require('.');

describe('basic', () => {
  let component;
  beforeEach(() => component = shallow(h(Component, { store: { tasks: [] } })));
  it('snapshot', () => expect(component.output()).toMatchSnapshot());
});

describe('User interaction', () => {
  it('Adds a task', () => {
    const store = { tasks: [] };
    const component = deep(h(Component, { store }))
    const form = component.find('.newTaskInput')
    const text = 'test'
    const chained = chain(() => () => {}, { value: () => text });
    form.simulate('submit', chained);
    expect(store).toMatchObject({ tasks: [{ text }] })
  });
  it('Deletes a task', () => {
    const store = { tasks: [{ text: 'test' }] };
    const component = deep(h(Component, { store }))
    const remove = component.find('.remove')
    remove.simulate('click');
    expect(store).toMatchObject({ tasks: [] })
  });
  it('Marks a task as checked', () => {
    const store = { tasks: [{ text: 'test' }] };
    const component = deep(h(Component, { store }))
    const checkbox = component.find('.checkbox').find('input');
    checkbox.simulate('change', { target: { checked: true } });
    expect(store).toMatchObject({ tasks: [{ text: 'test', completed: true }] });
  });
  it('Marks all tasks as checked', () => {
    const store = { tasks: [{ text: 'test' }] };
    const component = deep(h(Component, { store }))
    const checkAll = component.find('#checkAll');
    checkAll.simulate('change', { target: { checked: true } });
    expect(store).toMatchObject({ tasks: [{ text: 'test', completed: true }] });
  });
  it('Double click makes task editable', () => {
    const store = { tasks: [{ text: 'test' }] };
    const component = deep(h(Component, { store }))
    const taskUneditableBox = component
      .find('.taskWrapper')
      .find('.task')
      .find('.text');
    // console.log(`taskUneditableBox:`, taskUneditableBox);
    taskUneditableBox.simulate('dblclick');
    expect(store).toMatchObject({ tasks: [{ text: 'test', editing: true }] });
  });
  it('Double click makes task editable', () => {
    const store = { tasks: [{ text: 'test' }] };
    const component = deep(h(Component, { store }))
    const taskUneditableBox = component
      .find('.taskWrapper')
      .find('.task')
      .find('.text');
    // console.log(`taskUneditableBox:`, taskUneditableBox);
    taskUneditableBox.simulate('dblclick');
    expect(store).toMatchObject({ tasks: [{ text: 'test', editing: true }] });
  });
});
