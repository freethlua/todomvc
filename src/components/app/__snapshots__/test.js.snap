// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`basic snapshot 1`] = `
<div class="view-all container div">
  <div class="heading div">
    <div class="header div">todos</div>
  </div>
  <div class="tasks div">
    <div class="header div">
      <div class="checkAll div">
        <div class="div">
          <input
            id="checkAll"
            type="checkbox"
            checked={0}
            onchange={[Function onchange]}
            class="input"
           />
          <label
            for="checkAll"
            class="label"
          >
          </label>
        </div>
      </div>
      <div class="input div">
        <form
          onsubmit={[Function onsubmit]}
          class="newTaskInput form"
        >
          <input
            value=""
            placeholder="What needs to be done?"
            class="input"
           />
        </form>
      </div>
    </div>
    <div class="taskList div"></div>
  </div>
  <div class="footer div">
    <div class="footer div">
      <div class="itemsLeft div">0 items left</div>
      <div class="toggleView div">
        <div class="div">
          <input
            type="radio"
            name="toggle-view"
            value="all"
            id="toggle-view-all"
            onclick={[Function onclick]}
            checked={true}
            class="input"
           />
          <label
            for="toggle-view-all"
            class="label"
          >
            all
          </label>
        </div>
        <div class="div">
          <input
            type="radio"
            name="toggle-view"
            value="active"
            id="toggle-view-active"
            onclick={[Function onclick]}
            checked={false}
            class="input"
           />
          <label
            for="toggle-view-active"
            class="label"
          >
            active
          </label>
        </div>
        <div class="div">
          <input
            type="radio"
            name="toggle-view"
            value="completed"
            id="toggle-view-completed"
            onclick={[Function onclick]}
            checked={false}
            class="input"
           />
          <label
            for="toggle-view-completed"
            class="label"
          >
            completed
          </label>
        </div>
      </div>
      <div class="hidden div">
        <button
          onclick={[Function onclick]}
          class="clearCompleted button"
        >
          Clear completed
        </button>
      </div>
    </div>
  </div>
</div>
`;
